pkg/cloudprovider/providers/gce/kms.go

```diff
@@ -165,3 +181,16 @@ func unrecoverableCreationError(err error) bool {
 	}
 	return false
 }
+
+// getKeyVersionFromName parses the key version from the provided full path of the key,
+// as returned by cloudkms API.
+func getKeyVersionFromName(name string) (string, error) {
```

Comment:
this looks fairly brittle. Is it worth saving a few bytes per secret and having this ugly parsing code? Are there situations where having the whole string would help debugging?

If we want this code it should have a test.

Reply:
Indeed, it is brittle.
The information in the remaining part of the string is redundant. It either has constants for readability (`/projects/`) or information about the key which defines the key being used for encryption. Adding tests is slightly worse, since the ideal test for this would have to be an integration test, which would require the test account to have Google Cloud KMS access (might take long).
Suggesting an alternative in the comments.